### PR TITLE
Setting Markdown-it as markdown main render.

### PR DESCRIPTION
**Pull request by reyesrico**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,10 +25,11 @@
   "author": "Microsoft Corp",
   "license": "MIT",
   "dependencies": {
+    "@types/react-redux": "^4.4.38",
+    "@types/redux": "^3.6.0",
     "botframework-directlinejs": "^0.9.6",
     "core-js": "^2.4.1",
-    "he": "^1.1.1",
-    "marked": "^0.3.6",
+    "markdown-it": "^8.3.1",
```
Comment (danmarshall) [inline package.json]: is there a @types/markdown-it ?

File: src/FormattedText.tsx [hunk 2]
```diff
@@ -29,206 +53,12 @@ const renderPlainText = (text: string) => {
...
-        return this.addElement(<img key={this.key++} onLoad={ () => this.onImageLoad() } {...{ src: href, title, alt: text }} />);
-    }
-
-    text(text: string): string {
-        return this.addElement(<span key={this.key++}>{He.unescape(text)}</span>);
-    }
-}
+    markdownOptions = markdownOptions == null ? {html: true, linkify: true, typographer: true} : markdownOptions; 
+    const md = new MarkdownIt(markdownOptions);
+    const elements = md.render(src);
```
Comment (danmarshall) [inline src/FormattedText.tsx]: rename to html, since it isn't a collection of elements

File: src/FormattedText.tsx [hunk 3]
```diff
@@ -29,206 +53,12 @@ const renderPlainText = (text: string) => {
...
-            return '';
-        return this.addElement(<img key={this.key++} onLoad={ () => this.onImageLoad() } {...{ src: href, title, alt: text }} />);
-    }
-
-    text(text: string): string {
-        return this.addElement(<span key={this.key++}>{He.unescape(text)}</span>);
-    }
-}
+    markdownOptions = markdownOptions == null ? {html: true, linkify: true, typographer: true} : markdownOptions; 
+    const md = new MarkdownIt(markdownOptions);
```
Comment (danmarshall) [inline src/FormattedText.tsx]: can MarkdownIt be a singleton, instead of a new one for every render ?

File: src/FormattedText.tsx [hunk 4]
```diff
@@ -1,12 +1,36 @@
-import * as Marked from 'marked';
+import * as MarkdownIt from 'markdown-it';
 import * as React from 'react';
-import * as He from 'he';
+
+export interface MarkdownItOptions{
```
Comment (eanders-MS) [inline src/FormattedText.tsx]: Add a space before {

File: test/uitest1024.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+"use strict";
```
Comment (danmarshall) [inline test/uitest1024.js]: we need to work out these copies

File: src/FormattedText.tsx (hunk 4 above)
Comment (eanders-MS) [inline src/FormattedText.tsx]: No need to define this type. It's already in @types/markdown-it: https://www.npmjs.com/package/@types/markdown-it

File: test/commands_map.ts [hunk 6]
```diff
@@ -41,58 +41,51 @@ var commands_map: CommandValuesMap = {
...
+            setTimeout(() => {
+                right_arrow.click();
+                setTimeout(() => {
+                    right_arrow.click();
+                    setTimeout(() => {
+                        resolve(right_arrow.getAttribute('disabled') != null);
+                    }, 2000);
+                }, 500);
+            }, 500);
+        }),
```
Comment (eanders-MS) [inline test/commands_map.ts]: This seems unrelated. Is this part of the markdown-it changes?

File: test/commands_map.ts [hunk 7]
```diff
@@ -108,25 +101,23 @@ var commands_map: CommandValuesMap = {
...
+                    var element = document.querySelectorAll('.wc-carousel-item')[i];
+                    element.scrollIntoView();
             }
-
-            return right_arrow.getAttribute('disabled') != null;
-        },
+            setTimeout(() => {
+                resolve(right_arrow.getAttribute('disabled') != null);
+            }, 500);
+        }),
```
Comment (eanders-MS) [inline test/commands_map.ts]: I think you may need to merge latest then reapply your markdown changes to isolate your changes from others in this changeset.

File: test/uitest1024.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+				}
+			}
+			return results;
+		}
+
+		vo(testAllCommands)(function (err, results) {
+			done();
+		});
+	});
+});
```
Comment (eanders-MS) [inline test/uitest1024.js]: These changes also seem unrelated. Can you split into more targeted PRs?

File: package.json (hunk 1 above)
Comment (reyesrico) [inline package.json]: yes it is! I will include it for the next PR.

File: src/FormattedText.tsx (hunk 3 above)
Comment (reyesrico) [inline src/FormattedText.tsx]: Yes it can be a singleton. It will be included for the next PR.

File: test/commands_map.ts (hunk 6 above)
Comment (reyesrico) [inline test/commands_map.ts]: This is unrelated for markdown-it. Some tests were affected due the change of the library and the using of async / await words.

File: test/commands_map.ts (hunk 7 above)
Comment (reyesrico) [inline test/commands_map.ts]: I will do for the next PR.
